flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableFsDataOutputStream.java

```diff
@@ -360,4 +366,121 @@ private static boolean waitUntilLeaseIsRevoked(final FileSystem fs, final Path p
         }
         return isClosed;
     }
+
+    
+	/*
+	 * Run the dfs recover lease. recoverLease is asynchronous. It returns: -false when it starts the lease recovery (i.e. lease recovery not *yet* done) - true when the lease recovery has
+	 * succeeded or the file is closed.
+	 *
+	 * But, we have to be careful. Each time we call recoverLease, it starts the recover lease process over from the beginning. We could put ourselves in a situation
+	 * where we are doing nothing but starting a recovery, interrupting it to start again, and so on.
+	 *
+	 * The namenode will try to recover the lease on the file's primary node. If all is well, it should return near immediately.
+	 * But, as is common, it is the very primary node that has crashed and so the namenode will be stuck waiting on a socket timeout before it will ask another datanode to start the recovery.
+	 * It does not help if we call recoverLease in the meantime and in particular, subsequent to the socket timeout, a recoverLease invocation will cause us to start over from square one
+	 * (possibly waiting on socket timeout against primary node).
+	 * So, in the below, we do the following:
+	 * 1. Call recoverLease.
+	 * 2. If it returns true, break.
+	 * 3. If it returns false, wait a few seconds and then call it again.
+	 * 4. If it returns true, break.
+	 * 5. If it returns false, wait for what we think the datanode socket timeout is (configurable) and then try again.
+	 * 6. If it returns true, break.
+	 * 7. If it returns false, repeat starting at step 5. above. If HDFS-4525 is available, call it every second and we might be able to exit early.
+	 */
+	private static boolean recoverLease(Path path, DistributedFileSystem dfs) throws IOException {
```

Comment:
Looks like this new method is unused. Shouldn't we replace #waitUntilLeaseIsRevoked with it?